### PR TITLE
test: add file-based SQL formatter testing framework

### PR DESCRIPTION
**Pull request by shreyastelkar**
## Summary
image

- **Adds a fixture-driven testing framework** for `processDiffOutput` (the SQL formatter's diff-to-TextEdit engine)
- Tests verify the full pipeline end-to-end: `input.sql` + `sqlfmt.diff` → `processDiffOutput()` → TextEdits → apply edits → compare against `expected.sql`
- **Extracts `processDiffOutput`** from a private class method to an exported module-level function (pure structural refactor, identical algorithm) to enable direct unit testing
- **8 fixture sets** covering: issue #1717 long lines, basic reformatting, line splitting, multi-chunk changes, no-op, Jinja templates, blank line deletion, trailing newlines
- Marks `multi-chunk-changes` fixture as a known bug with `it.failing` (will be fixed by PR #1789)

## What's included

| Component | Description |
|---|---|
| `src/test/helpers/applyTextEdits.ts` | TextEdit application engine simulating VSCode's simultaneous edit semantics |
| `src/test/helpers/fixtureLoader.ts` | Auto-discovers fixture directories under `src/test/fixtures/formatter/` |
| `src/test/fixtures/formatter/` | 8 fixture directories, each with `input.sql`, `expected.sql`, `sqlfmt.diff` |
| `src/test/fixtures/formatter/README.md` | Documents how to regenerate fixtures with `shandy-sqlfmt` |
| `src/test/suite/dbtDocumentFormattingEditProvider.test.ts` | 11 parameterized + edge case tests |
| `src/test/setup.ts` + `src/test/mock/vscode.ts` | Added `TextEdit` mock |
| `dbtDocumentFormattingEditProvider.ts` | Extract `processDiffOutput` as exported function (no algorithm change) |

## Why this matters

The formatter has had multiple bugs (#1717 and others). Previous tests only verified that TextEdit objects were produced — they never verified that **applying** those edits to the input actually produced the correct output. This framework catches regressions by testing the complete pipeline.

## Relationship to PR #1789

This PR provides the **testing infrastructure**. PR #1789 will fix the `processDiffOutput` algorithm itself. Once #1789 lands, the `multi-chunk-changes` `it.failing` test will start passing, signaling removal of the `.failing` marker.

## Test plan

- [x] All 147 tests pass (146 passed + 1 skipped)
- [x] 7 fixture tests verify correct end-to-end output
- [x] 1 `it.failing` test confirms the multi-chunk bug exists
- [x] 3 edge case tests (empty diff, summary-only diff, no-newline marker)
- [x] Existing 136 tests unaffected

🤖 Generated with [Claude Code](https://claude.com/claude-code)

----

> [!IMPORTANT]
> Adds a fixture-driven testing framework for `processDiffOutput`, extracting it for testability, and includes various test cases and fixtures.
> 
>   - **Testing Framework**:
>     - Adds fixture-driven testing framework for `processDiffOutput` in `dbtDocumentFormattingEditProvider.ts`.
>     - Tests cover end-to-end pipeline: `input.sql` + `sqlfmt.diff` → `processDiffOutput()` → TextEdits → apply edits → compare with `expected.sql`.
>     - 8 fixture sets for various cases: long lines, basic reformatting, line splitting, multi-chunk changes, no-op, Jinja templates, blank line deletion, trailing newlines.
>     - Marks `multi-chunk-changes` fixture as a known bug with `it.failing`.
>   - **Function Extraction**:
>     - Extracts `processDiffOutput` from a private class method to an exported function in `dbtDocumentFormattingEditProvider.ts` for direct testing.
>   - **Helper Functions**:
>     - Adds `applyTextEdits.ts` to simulate VSCode's TextEdit application.
>     - Adds `fixtureLoader.ts` to auto-discover fixture directories.
>   - **Mock and Setup**:
>     - Updates `mock/vscode.ts` and `setup.ts` to include `TextEdit` mock.
>   - **Tests**:
>     - Adds `dbtDocumentFormattingEditProvider.test.ts` with 11 parameterized and edge case tests.
>     - Includes edge case tests for empty diff, summary-only diff, and no-newline marker.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=AltimateAI%2Fvscode-dbt-power-user&utm_source=github&utm_medium=referral)<sup> for acc0551c40b68a3175b0b24a25f362f01182ec52. You can [customize](https://app.ellipsis.dev/AltimateAI/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive SQL formatter fixtures for trailing-newline, reformatting, long-line, Jinja, multi-CTE and other edge cases.
  * Added test helpers for loading fixtures and applying text edits, a mocked TextEdit API for tests, and a new test suite validating document formatting edits.

* **Documentation**
  * Added a README describing fixture structure, test flow, and regeneration commands.

### CODE REVIEW

File: src/test/integration/formatter.test.ts [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+            "sqlfmt formatted output should equal input when no changes needed",
+          );
+        });
+        continue;
+      }
+
+      if (KNOWN_BUGGY_FIXTURES.has(fixtureName)) {
+        // Skip known buggy fixtures — will be fixed by PR #1789
+        test.skip(`${fixtureName}: known bug, fix expected in PR #1789`, function () {});
+        continue;
```
Comment (shreyastelkar) [inline src/test/integration/formatter.test.ts]: This should be removed.
